### DISCUSSION TITLE
Quick and Dirty Setup for Ollama, Open-WebUI, Docker - Slow, CPU-Based Setup

### DISCUSSION THREAD

**Discussion by jsun1973** [open-webui/open-webui#8003] [General]
I'm sharing this because it is really simple, but for some reason not readily handy to get a local version in Docker working where I could use a web browser locally on the network with ubuntu server running docker and managing Ollama and Open-WebUI in seperate containers. Here are my notes and what I did to get it working. Feel free to try them:

Create a container and download ollama. Create persistence:

`docker run -d  --name ollama  -p 11434:11434   -v ollama_volume:/root/.ollama ollama/ollama:latest`

Check to ensure the container was created:

`docker ps`

Create container and download Open-WebUI:

`docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v open-webui:/app/backend/data --name open-webui --restart always ghcr.io/open-webui/open-webui:main`

Pull llama3.2 into ollama and test run:

`docker exec -it ollama ollama run llama3.2`

To exit, type `/bye`

Connecting using browser:

`[server IP]:3000`

Keep in mind this is just a basic instance. It is slow compared to GPU-based LLM's.

**Comment by ghost** [comment]


**Reply by jsun1973** [reply]
Thank you! I will give those a shot the next time I'm in front of the machine. Appreciate it!

**Reply by jsun1973** [reply]
I tried it and it seems to have helped!  Thanks for the suggestion!
